From 8ae71c99ce763f9ce49afe25f5569931dfd6cd1e Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Thu, 28 Jul 2016 13:24:52 +0500
Subject: [PATCH] init-install.sh: /etc/mtab make a softlink rather than a copy

Using a copy would only make management of devices erroneous
and makes the system unstable in some scenarios as tools will
have to manipulate both files separately. A link ensures that
both files /proc/mounts and /etc/mtab will have the same
information at all times and this is how it is handled
on newer systems where there is such a need.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 init-install.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git init-install.sh init-install.sh
index 1b69b71..5b8f925 100644
--- init-install.sh
+++ init-install.sh
@@ -122,8 +122,8 @@ if [ ! -b /dev/loop0 ] ; then
 fi
 
 mkdir -p /tmp
-if [ ! -L /etc/mtab ]; then
-    cat /proc/mounts > /etc/mtab
+if [ ! -L /etc/mtab ] && [ -e /proc/mounts ]; then
+    ln -sf /proc/mounts /etc/mtab
 fi
 
 disk_size=$(parted ${device} unit mb print | grep Disk | cut -d" " -f 3 | sed -e "s/MB//")
-- 
1.9.1

